opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java

```diff
@@ -189,48 +195,59 @@ public void setupHttpClient() {
-            } else {
-                if (statusCode == HttpStatus.SC_UNAUTHORIZED) {
-                    logError("Invalid credentials for: " + userName + " using " + requestURL);
-                    loginResponse = UNAUTHORIZED;
-                } else {
-                    logError("Bad response from Dristhi. Status code:  " + statusCode + " username: "
-                            + userName + " using " + requestURL);
-                    loginResponse = UNKNOWN_RESPONSE;
-                }
-
-                String responseString = getResponseString(httpResponse);
+            } else if (statusCode == HttpStatus.SC_UNAUTHORIZED) {
+                logError("Invalid credentials for: " + userName + " using " + requestURL);
+                loginResponse = UNAUTHORIZED;
+            } else if (StringUtils.isNotBlank(responseString)) {
+                responseString = StringUtils.substringBetween(responseString, "<p><b>message</b>", "</u></p>");
                 if (StringUtils.isNotBlank(responseString)) {
-                    responseString = StringUtils.substringBetween(responseString, "<p><b>message</b>", "</u></p>");
-                    if (StringUtils.isNotBlank(responseString)) {
-                        responseString = responseString.replace("<u>", "").trim();
-                        loginResponse = CUSTOM_SERVER_RESPONSE.withMessage(responseString);
-                    }
+                    responseString = responseString.replace("<u>", "").trim();
+                    loginResponse = CUSTOM_SERVER_RESPONSE.withMessage(responseString);
                 }
+            } else {
+                logError("Bad response from Dristhi. Status code:  " + statusCode + " username: "
+                        + userName + " using " + requestURL);
+                loginResponse = UNKNOWN_RESPONSE;
             }
-        } catch (ConnectionPoolTimeoutException | SocketTimeoutException e) {
-            Log.e(TAG, e.getMessage(), e);
+        } catch (MalformedURLException e) {
+            Log.e(TAG, "Failed to check credentials bad url " + requestURL, e);
+            loginResponse = MALFORMED_URL;
+        } catch (SocketTimeoutException e) {
+            Log.e(TAG, "SocketTimeoutException when authenticating " + userName, e);
             loginResponse = TIMEOUT;
+            Log.e(TAG, "Failed to check credentials of: " + userName + " using " + requestURL + ". "
+                    + "" + "" + "Error: " + e.toString(), e);
         } catch (IOException e) {
             Log.e(TAG, "Failed to check credentials of: " + userName + " using " + requestURL + ". "
                     + "" + "" + "Error: " + e.toString(), e);
             loginResponse = NO_INTERNET_CONNECTIVITY;
-        } catch (IllegalArgumentException e) {
-            Log.e(TAG, e.getMessage(), e);
-            loginResponse = MALFORMED_URL;
         } finally {
-            httpClient.consumeResponse(httpResponse);
+            assert urlConnection != null;
```

Comment:
Should assert be used in production code?

Why use assert instead of `if` statement?